pyartcd/pyartcd/pipelines/prepare_release.py

```diff
@@ -156,62 +159,49 @@ async def run(self):
-                                release_date += timedelta(days=1)
-                            release_date = release_date.strftime("%Y-%b-%d")
-                        advisories[ad] = self.create_advisory(advisory_type=advisory_type,
-                                                              art_advisory_key=ad,
-                                                              release_date=release_date)
-                        await self._slack_client.say_in_thread(
-                            f"{ad} advisory created with release date {release_date}")
-                        continue
+                if advisories[ad] >= 0:
+                    continue
+                if ad in ["advance", "prerelease"]:
+                    release_date = None
+                    if ad == "advance":
+                        # Set release date of advance advisory to one week before GA
+                        # Eg one week before '2024-Feb-07' should be '2024-Jan-31'
+                        ga_date = datetime.strptime(self.release_date, "%Y-%b-%d")
+                        one_week_before = ga_date - timedelta(days=7)
+                        release_date = one_week_before.strftime("%Y-%b-%d")
+                    elif ad == "prerelease":
+                        # Set release date of prerelease advisory to 3 days after when we prepare the release
+                        # it should not be a weekend
+                        today = datetime.now(tz=timezone.utc)
+                        release_date = today + timedelta(days=3)
+                        if release_date.weekday() == 5:  # Saturday
+                            release_date += timedelta(days=2)
+                        elif release_date.weekday() == 6:  # Sunday
+                            release_date += timedelta(days=1)
+                        release_date = release_date.strftime("%Y-%b-%d")
                     advisories[ad] = self.create_advisory(advisory_type=advisory_type,
                                                           art_advisory_key=ad,
-                                                          release_date=self.release_date,
-                                                          batch_id=batch["id"] if batch else 0)
+                                                          release_date=release_date)
+                    await self._slack_client.say_in_thread(
+                        f"{ad} advisory created with release date {release_date}")
+                    continue
+                batch_id = 0
+                if batch and ad != "microshift":
+                    # Set batch_id for advisories other than microshift
+                    batch_id = int(batch["id"])
+                    # Ensure that the batch is unlocked
+                    batch = await self._ensure_batch_status(self._errata_api, batch, lock=False, dry_run=self.dry_run)
+                advisories[ad] = self.create_advisory(advisory_type=advisory_type,
```

Comment:
This makes me think if we should have a lightweight Advisory and Batch class and then it can internally call `batch.unlock()` before `create()`. Strictly optional suggestion/thought